### DISCUSSION TITLE
New handler (onRepliedMessage)

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#262] [Ideas]
Hello.

In the bot that I am designing, users sends message to the administrator from the «Support» section. In order to reply to the users, there must be an `onRepliedMessage` event...

![image](https://user-images.githubusercontent.com/78247242/202141501-fc436340-1dd1-4c0b-9f84-3834d195299e.png)

example:
```php
$bot->onRepliedMessage(function (Nutgram $bot) {
    // some code here
    // example: detect and answer to user's message
});

// ============================

// another example:
$bot->onRepliedMessage(MessageTypes::PHOTO, function (Nutgram $bot) {
    // some code here
});

// etc...
```

How can I implement this? 🤔

I don't know if this function (feature) is good or not... but since there is no such thing, I have implemented it in the `fallbackOn` handler to detect these types of messages :(

**Comment by Lukasss93** [comment]
I suggest you to use a Conversation: https://nutgram.dev/docs/usage/conversations

**Reply by ghost** [reply]
Yes. But this is the general situation. It is checked when the user is not in any conversation...

In this case, we check the replied message and respond accordingly

**Comment by sergix44** [comment]
Its just an onMessage with an `if` the message contains a replied message.
Makes sense adding shortcuts and for the main type, but for sure I'm not going to add a listener for every type of of subtypes of every update. Thats just too much.
